sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/config/CosmosConfig.java

```diff
@@ -17,32 +17,39 @@ public class CosmosConfig {
 
     private final boolean queryMetricsEnabled;
 
+    private final int maxDegreeOfParallelism;
+
     /**
      * Initialization
      *
      * @param responseDiagnosticsProcessor must not be {@literal null}
      * @param queryMetricsEnabled must not be {@literal null}
+     * @param maxDegreeOfParallelism must not be {@literal null}
      */
-    @ConstructorProperties({"responseDiagnosticsProcessor", "queryMetricsEnabled"})
+    @ConstructorProperties({"responseDiagnosticsProcessor", "queryMetricsEnabled", "maxDegreeOfParallelism"})
```

Comment:
Changing an existing constructor is a breaking change. We should create a new constructor which will then take this property in addition to other properties. 
If there are too many constructors, we should think about the builder pattern to support this.